examples/gnrc_tcp_cli/main.c

```diff
@@ -0,0 +1,235 @@
+            bufs[tid][i+1] = 0x00;
+        }
+
+        /* Send Data, stop if errors were found */
+        for (sent = 0; sent < sizeof(bufs[tid]) && ret >= 0; sent += ret) {
+            ret = gnrc_tcp_send(&tcb, bufs[tid] + sent, sizeof(bufs[tid]) - sent, 0);
+            switch (ret) {
+                case -ENOTCONN:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ENOTCONN\n", tid);
+                    break;
+
+                case -ECONNABORTED:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ECONNABORTED\n", tid);
+                    break;
+
+                case -ETIMEDOUT:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ETIMEDOUT\n", tid);
+                    break;
+
+                case -ECONNRESET:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ECONNRESET\n", tid);
+                    break;
+
+                default:
+                    if (ret >= 0) {
+                        DEBUG("TID=%"PRIi32" : gnrc_tcp_send() : %"PRIi32" Bytes sent\n", tid, ret);
+                    }
+                    else {
+                        printf("TID=%"PRIi32" : gnrc_tcp_send() : %"PRIi32"\n", tid, ret);
+                        break;
+                    }
+              }
+        }
+
+        /* Receive Data, stop if errors were found */
+        for (rcvd = 0; rcvd < sizeof(bufs[tid]) && ret >= 0; rcvd += ret) {
+            ret = gnrc_tcp_recv(&tcb, (void *)(bufs[tid] + rcvd), sizeof(bufs[tid]) - rcvd,
+                                GNRC_TCP_CONNECTION_TIMEOUT_DURATION);
+            switch (ret) {
+                case -ENOTCONN:
+                    printf("TID=%"PRIi32" : gnrc_tcp_rcvd() : -ENOTCONN\n", tid);
+                    break;
+
+                case -EAGAIN:
+                    printf("TID=%"PRIi32" : gnrc_tcp_rcvd() : -EAGAIN\n", tid);
+                    ret = 0;
+                    break;;
+
+                case -ECONNABORTED:
+                    printf("TID=%"PRIi32" : gnrc_tcp_rcvd() : -ECONNABORTED\n", tid);
+                    break;
+
+                case -ECONNRESET:
+                    printf("TID=%"PRIi32" : gnrc_tcp_rcvd() : -ECONNRESET\n", tid);
+                    break;
+
+                case -ETIMEDOUT:
+                    printf("TID=%"PRIi32" : gnrc_tcp_rcvd() : -ETIMEDOUT\n", tid);
+                    break;
+
+                default:
+                    if (ret >= 0) {
+                        DEBUG("TID=%"PRIi32" : gnrc_tcp_rcvd() : %"PRIi32" Bytes read.\n", tid, ret, ret);
+                    }
+                    else {
+                        printf("TID=%"PRIi32" : gnrc_tcp_rcvd() : %"PRIi32"\n", tid, ret);
+                        break;
+                    }
+              }
+        }
+
+        /* If there was no error: Check received pattern */
+        if(ret < 0){
+            for (uint32_t i=0; i < sizeof(bufs[tid]) - 1; i+= 2) {
+                if (bufs[tid][i] != 0xAA || bufs[tid][i+1] != 0x77) {
```

Comment:
I think these patterns should be configurable by `#defines` at the top, maybe even via Makefile, too